listener/listener_test.go

```diff
@@ -101,36 +54,58 @@ func TestListener(t *testing.T) {
 		So(err, ShouldBeNil)
 		So(len(shards), ShouldEqual, 1)
 
-		l, err := NewListener(stream, shards[0], func(c *Config) {
+		p, err := producer.NewProducer(func(c *producer.Config) {
```

Comment:
If we create a producer, we don't need putRecord above.

It may be simpler to keep putRecord instead of creating a Producer, as producers have an inherent delay due to their batch sizes.